runtime/builtin_types.go

```diff
@@ -631,15 +631,39 @@ func builtinRound(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 			return nil, raised
 		}
 	}
-	// TODO: implement this.
-	if ndigits != 0 {
-		return nil, f.RaiseType(NotImplementedErrorType, "round with ndigits is not implemented in grumpy")
-	}
 	number, isFloat := floatCoerce(args[0])
+
 	if !isFloat {
 		return nil, f.RaiseType(TypeErrorType, "a float is required")
 	}
-	return NewFloat(math.Floor(number + 0.5)).ToObject(), nil
+
+	if math.IsNaN(number) || math.IsInf(number, 0) || number == 0.0 {
+		return NewFloat(number).ToObject(), nil
+	}
+
+	var result float64
+	if ndigits == 0 {
+		result = round(number, ndigits)
+		// round(12.5) => 13, round(-12.5) => -13
```

Comment:
I guess I still don't understand why we need all the special cases. ISTM that something like this will work for all cases:

```
neg := false
if number < 0 {
	neg = true
	number = -number
}
pow := math.Pow(10.0, float64(ndigits))
result = math.Floor(number*pow+0.5) / pow
if neg {
	result = -result
}
```